presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -191,11 +191,13 @@ public static Domain getDomain(Type type, long rowCount, Statistics<?> statistic
                 failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, floatStatistics);
                 return Domain.create(ValueSet.all(type), hasNullValue);
             }
+            if (floatStatistics.genericGetMin().isNaN() || floatStatistics.genericGetMax().isNaN()) {
+                return Domain.create(ValueSet.all(type), hasNullValue);
+            }
 
             ParquetIntegerStatistics parquetStatistics = new ParquetIntegerStatistics(
-                    (long) floatToRawIntBits(floatStatistics.getMin()),
-                    (long) floatToRawIntBits(floatStatistics.getMax()));
-
+                    (long) floatToIntBits(floatStatistics.getMin()),
```

Comment:
why the switch? Should not matter as for non-NaNs behaviour is the same. Yet I am curious.

Reply:
Simply because we know that the value isn't `NaN`.  It isn't a necessary change, so I can roll it back.

Reply:
Actually if we know that value is *not NaN* calling to `floatToRawIntBits` seems more natural. As this is `floatToIntBits` which has NaN handling.

Here is code of floatToIntBits():
```
    public static int floatToIntBits(float value) {
        if (!isNaN(value)) {
            return floatToRawIntBits(value);
        }
        return 0x7fc00000;
    }
```

Reply:
Thanks, I got my wires crossed there.